### PR TITLE
Added auth file documentation

### PR DESCRIPTION
**Pull request by mcardosos**
PTAL
@jhendrixMSFT @marstr 
Based on [python's docs](https://docs.microsoft.com/en-us/python/azure/python-sdk-azure-authenticate?view=azure-python#mgmt-auth-file).

### CODE REVIEW

File: arm/README.md [hunk 1]
```diff
@@ -91,6 +91,58 @@ post,
...
 
+### Use an authentication file
+
+This SDK also supports authentication with a JSON file containing credentials for the service principal. In the Azure CLI, you can create a service principal and its authentication file with this command:
+
+``` bash
+az ad sp create-for-rbac --sdk-auth > mycredentials.json
+```
+
+THe file lloks like this, in case you want to create it yourself:
```
Comment (marstr) [inline arm/README.md]: Typo `THe`

File: arm/README.md [hunk 2]
```diff
@@ -91,6 +91,58 @@ post,
...
+
+``` json
+{
+    "clientId": "ad735158-65ca-11e7-ba4d-ecb1d756380e",
+    "clientSecret": "b70bb224-65ca-11e7-810c-ecb1d756380e",
+    "subscriptionId": "bfc42d3a-65ca-11e7-95cf-ecb1d756380e",
+    "tenantId": "c81da1d8-65ca-11e7-b1d1-ecb1d756380e",
```
Comment (marstr) [inline arm/README.md]: You should replace all of the values above with comments about what that value is.

`
    "clientSecret":"<whatever>"
`

File: arm/README.md [hunk 3]
```diff
@@ -91,6 +91,58 @@ post,
 [Developer's Guide to Auth with Azure Resource Manager API](http://www.dushyantgill.com/blog/2015/05/23/developers-guide-to-auth-with-azure-resource-manager-api/),
 that is also quite helpful.
 
+### Use an authentication file
```
Comment (marstr) [inline arm/README.md]: I'm not sure that the AzureRM README file is really the best place for this. Honestly, I think we're overdue to start a Wiki in our repository. We could have a page on authentication, and a page for the roadmap as mentioned in #804.

File: arm/README.md [hunk 4]
```diff
@@ -91,6 +91,58 @@ post,
...
+    "sqlManagementEndpointUrl": "https://management.core.windows.net:8443/",
+    "galleryEndpointUrl": "https://gallery.azure.com/",
+    "managementEndpointUrl": "https://management.core.windows.net/"
+}
+```
+
+Save this file in a secure location on your system where your code can read it. Set an environment variable with the full path to the file:
+
+``` bash
+export AZURE_AUTH_LOCATION=/secure/location/mycredentials.json
```
Comment (marstr) [inline arm/README.md]: I would mention this environment variable higher up as well. It is important that people know that unlike other examples of environment variables in our Samples and README, this one is referenced by our library. 

Also, we should add a PowerShell example as well.

Comment (mcardosos) [reply arm/README.md]: Fixed

File: arm/README.md (hunk 2 above)
Comment (mcardosos) [reply arm/README.md]: Fixed

File: arm/README.md (hunk 1 above)
Comment (mcardosos) [reply arm/README.md]: Fixed

**Pull request merged**